### DISCUSSION TITLE
BaseModel Validation can raise another error?

### DISCUSSION THREAD

**Discussion by tonglin96** [fastapi/fastapi#8100] [Questions] [answered]
i want to raise another error, maybe myself defined. or does it supports define a error in BaseModel what i want shows?

**Comment by tonglin96** [comment]
i want to define error msg that shows in response body for every fields

**Comment by pmav99** [comment, marked as answer]
@a372661369 This is a pydantic question, not a fastapi one. Check here for defining custom validators: https://pydantic-docs.helpmanual.io/#validators

UPD: working link https://docs.pydantic.dev/latest/concepts/validators/

**Comment by tonglin96** [comment]
@pmav99 thk

**Comment by dmontagu** [comment]
In addition to a pydantic question, you also need to figure out how to override the response for a fastapi `RequestValidationError`. I’m not sure how hard this is and haven’t seen an example of how to do this; if anyone has done this please share!

**Comment by tiangolo** [comment]
Thanks for the help and discussion here guys!

Here are the docs about how to override the default exception handlers: https://fastapi.tiangolo.com/tutorial/handling-errors/#override-the-default-exception-handlers
